src/main/java/org/stellar/sdk/FeeBumpTransaction.java

```diff
@@ -113,21 +120,18 @@ public static class Builder {
      * @param accountConverter The AccountConverter which will be used to encode the fee account.
      * @param inner The inner transaction which will be fee bumped.
      */
-    public Builder(AccountConverter accountConverter, Transaction inner) {
-      inner = checkNotNull(inner, "inner cannot be null");
+    public Builder(AccountConverter accountConverter, final Transaction inner) {
+      checkNotNull(inner, "inner cannot be null");
       EnvelopeType txType = inner.toEnvelopeXdr().getDiscriminant();
       this.mAccountConverter = checkNotNull(accountConverter, "accountConverter cannot be null");
       if (inner.toEnvelopeXdr().getDiscriminant() == EnvelopeType.ENVELOPE_TYPE_TX_V0) {
-        this.mInner = new Transaction(
-            inner.accountConverter,
-            inner.getSourceAccount(),
-            inner.getFee(),
-            inner.getSequenceNumber(),
-            inner.getOperations(),
-            inner.getMemo(),
-            inner.getTimeBounds(),
-            inner.getNetwork()
-        );
+        this.mInner = new TransactionBuilder(inner.accountConverter, new Account(inner.getSourceAccount(), inner.getSequenceNumber() - 1), inner.getNetwork())
+                .setBaseFee((int)inner.getFee())
+                .addOperations(Arrays.asList(inner.getOperations()))
+                .addMemo(inner.getMemo())
+                .addPreconditions(new TransactionPreconditions.TransactionPreconditionsBuilder().timeBounds(inner.getTimeBounds()).build())
```

Comment:
Can the `final Transaction inner` have other preconditions that need to be added here?

Reply:
it's a `V0` tx since it's in this block, so TimeBounds is the only precondition'able attribute available from that version of tx right?